data-raw/healthy-places/travel-time-to-gp.R

```diff
@@ -0,0 +1,233 @@
+library(tidyverse)
+library(geographr)
+library(osmdata)
+library(jsonlite)
+library(httr)
+library(sf)
+library(traveltimeR)
+library(readxl)
+
+Sys.setenv(TRAVELTIME_ID = "<INSERT YOUR ID HERE>")
+Sys.setenv(TRAVELTIME_KEY = "<INSERT YOUR API KEY HERE>")
+
+# ---- Fetch GPs in Northern Ireland ----
+# Bounding box for Northern Ireland
+ni_bb <-
+  getbb("Northern Ireland")
+
+# Search for GPs
+GPs <- opq(ni_bb, timeout = 1000) %>%
+  add_osm_feature("amenity", "doctors") %>%
+  osmdata_sf()
+
+# Get Local Authorities in Great Britain for the next step
+ni_lad <-
+  boundaries_ltla21 |>
+  filter(str_detect(ltla21_code, "^N"))
+
+# Some GPs are not in Northern Ireland - remove them from the dataset
+ni_GPs <-
+  GPs$osm_points[ni_lad, ] |>
+  select(osm_id)
```

Comment:
The `GPs` dataset also contains 77 polygons and 2 multipolygons. I scanned the osmdata package docs and tried pritning these polygons with `ggplot(GPs$osm_polygons) + geom_sf(colour = "black", fill = NA)`, but can't figure out what they are.

Are you sure they can be dropped?

Reply:
Yes - those points are not in Northern Ireland.

This code plots their locations:

```r
ni_disjoint <- 
  GPs$osm_points[ni_lad, , op = st_disjoint] |>
  select(osm_id)

# - Test plot to check GP locations -
ni_lad |>
  ggplot() +
  geom_sf(
    fill = NA,
    colour = "black"
  ) +
  geom_sf(
    data = ni_disjoint,
    inherit.aes = FALSE,
    colour = "red"
  ) +
  geom_sf(
    data = ni_GPs,
    inherit.aes = FALSE,
    colour = "blue"
  )
```